### PR TITLE
Implemented tasks for week2

### PR DESCRIPTION
**Pull request by bhumikama**
Implemented these tasks for week2 

1. manipulate the movie array, implementing these functions:
   a function for searching for a provided keyword in the movie title;
   a function for sorting the movie array by one or a few selected properties.

### CODE REVIEW

File: movie.js [hunk 1]
```diff
@@ -5,17 +5,29 @@ function createMovieCard(movie) {
...
+          movie.rating >= 1 ? "&#9733;" : "&#9734;"
+        }</button> 
+        <button class="star">${
+          movie.rating >= 2 ? "&#9733;" : "&#9734;"
+        }</button>
+        <button class="star">${
+          movie.rating >= 3 ? "&#9733;" : "&#9734;"
+        }</button>
+        <button class="star">${
+          movie.rating >= 1 ? "&#9733;" : "&#9734;"
```
Comment (vdaguenet) [inline movie.js]:
```suggestion
          movie.rating >= 4 ? "&#9733;" : "&#9734;"
```

File: movie.js [hunk 2]
```diff
@@ -5,17 +5,29 @@ function createMovieCard(movie) {
...
+        }</button>
+        <button class="star">${
+          movie.rating >= 1 ? "&#9733;" : "&#9734;"
```
Comment (vdaguenet) [inline movie.js]:
```suggestion
          movie.rating >= 5 ? "&#9733;" : "&#9734;"
```

File: movie.js [hunk 3]
```diff
@@ -124,12 +144,75 @@ function displayMovies(movieList) {
...
+  const lowerCaseKeyword = keyword.toLowerCase();
+  const filteredMovies = movies.filter((movie) =>
+    movie.title.toLowerCase().includes(lowerCaseKeyword)
+  );
+
+  displayMovies(filteredMovies, keyword);
+}
+
+function sortMovies(movieList, sortBy) {
+  const sortedMovies = [...movieList]; //creates a shallow copy since sorting overwrites the original array
```
Comment (vdaguenet) [inline movie.js:175]: Yes!

File: movie.js [hunk 4]
```diff
@@ -124,12 +144,75 @@ function displayMovies(movieList) {
...
+
+  const closeIcon = document.querySelector(".search__close");
+  closeIcon.addEventListener("click", () => {
+    const searchInput = document.getElementById("search-input");
+    searchInput.value = "";
+    displayAllMovies();
+  });
+
+  const searchInput = document.getElementById("search-input");
+  searchInput.addEventListener("input", (event) => {
```
Comment (vdaguenet) [inline movie.js:206]: It's a good idea to listen to the `input` event in the search bar to filter movies. It makes the app quite reactive. However, displaying all the movies once the input is cleared would be good. For example:
1. I type "b" in the search bar. The app displays Barbie, Batman and Black Widow.
2. I erase the "b". The app still displays Barbie, Batman and Black Widow. I can not see the full list of movies anymore.

**Review by vdaguenet — APPROVED** [review]
Excellent work!

If you have some time and are up for the challenge, you can try to apply sorting on the filtered movies after a search. Let's say the search returns 3 movies. When the user selects to sort by year, the app displays these 3 movies sorted and does not go back to the full list of movies. **<ins>This is purely OPTIONAL as it was not part of the requirements for the week 2 assignment.</ins>**

File: movie.js [hunk 5]
```diff
@@ -124,12 +144,75 @@ function displayMovies(movieList) {
...
+    const keyword = event.target.value;
```
Comment (vdaguenet) [inline movie.js]: You should `trim()` this to avoid searching for spaces

**Comment by bhumikama** [comment]
@vdaguenet , thanks for the review and suggestions.
I have implemented both the optional task and changes you suggested.
Kindly have another look and let me know if I need to change anything.

Thank you

**Review by vdaguenet — APPROVED** [review]
Cool!

**Pull request merged**